testing/src/main/java/org/oppia/android/testing/ResetFeatureFlagToDefault.kt

```diff
@@ -0,0 +1,14 @@
+package org.oppia.android.testing
+
+import org.oppia.android.util.platformparameter.FeatureFlag
+
+/**
+ * Repeatable test method annotation for resetting the feature flag for tests of the
+ * specific method may run on. The feature flag names are provided by the
+ * FeatureFlagConstants.kt.
+ *
+ * Note that this annotation only works if the test also has an [OppiaTestRule] hooked up.
+ */
+@Target(AnnotationTarget.FUNCTION)
+@Repeatable
+annotation class ResetFeatureFlagToDefault(val name: FeatureFlag)
```

Comment:
Similar to my comment in ``EnableFeatureFlag``, cascading/duplication needs to be considered here. I'd expect any cases where an enable/disable happens and reset at the same level to be an error (i.e. a test method can only reset, not enable/disable and then reset).

For strictness, I'd also expect that reset throws an error if there's no explicit override of the feature at the class level (i.e. there's no reason to reset). While resetting a non-overridden flag is a no-op, allowing it to be checked in could result in false assumptions and a fundamental misunderstanding of what the test and surrounding test suite is actually doing.